dialogflow/cloud-client/src/test/java/com/example/dialogflow/KnowledgeBaseManagementIT.java

```diff
@@ -35,100 +47,127 @@
+        "FAQ",
+        "https://cloud.google.com/storage/docs/faq");
+    assertEquals(DOCUMENT_BASE_NAME, document.getDisplayName());
+
+    // List the Documents
+    List<Document> documents = DocumentManagement.listDocuments(knowledgeBaseName);
+    assertEquals(1, documents.size());
+    assertEquals(DOCUMENT_BASE_NAME, documents.get(0).getDisplayName());
 
-    // Detect Intent with Knowledge Base
-    detectIntentKnowledge.detectIntentKnowledge(PROJECT_ID, knowledgeBaseId,SESSION_ID,
-        LANGUAGE_CODE, TEXTS);
-    got = bout.toString();
-    assertThat(got).contains("Knowledge results");
+    // Get the Document
+    document = DocumentManagement.getDocument(document.getName());
+    assertEquals(DOCUMENT_BASE_NAME, document.getDisplayName());
 
     // Delete the Document
-    bout.reset();
-    documentManagement.deleteDocument(PROJECT_ID,knowledgeBaseId,documentId);
-    got = bout.toString();
-    assertThat(got).contains("The document has been deleted.");
+    DocumentManagement.deleteDocument(document.getName());
 
     // List the Document
-    documentManagement.listDocuments(PROJECT_ID,knowledgeBaseId);
-    got = bout.toString();
-    assertThat(got).doesNotContain("Display Name: " + DOCUMENT_BASE_NAME);
-
-    // Delete the Knowledge Base
-    knowledgeBaseManagement.deleteKnowledgeBase(PROJECT_ID,knowledgeBaseId);
-
-    // List Knowledge Bases
-    knowledgeBaseManagement.listKnowledgeBases(PROJECT_ID);
-    got = bout.toString();
-    assertThat(got).doesNotContain("Display Name: " + KNOWLEDGE_BASE_NAME);
-
+    documents = DocumentManagement.listDocuments(knowledgeBaseName);
+    assertEquals(0, documents.size());
   }
 
-}
\ No newline at end of file
+  @Test
+  public void testDetectIntentKnowledge() throws Exception {
+    // Create a Knowledge Base
+    KnowledgeBase knowledgeBase =
+        KnowledgeBaseManagement.createKnowledgeBase(PROJECT_ID, KNOWLEDGE_BASE_NAME);
+    String knowledgeBaseName = knowledgeBase.getName();
+
+    // Create a Document - one needs to exist in order for detectIntentKnowledge to provide answers.
+    Document document = DocumentManagement.createDocument(
+        knowledgeBaseName,
+        DOCUMENT_BASE_NAME,
+        "text/html",
+        "FAQ",
+        "https://cloud.google.com/storage/docs/faq");
+    assertEquals(DOCUMENT_BASE_NAME, document.getDisplayName());
+
+    Map<String, KnowledgeAnswers> allAnswers = DetectIntentKnowledge
+        .detectIntentKnowledge(PROJECT_ID, knowledgeBaseName, SESSION_ID, LANGUAGE_CODE, TEXTS);
+    assertEquals(1, allAnswers.size());
+    KnowledgeAnswers knowledgeAnswers = allAnswers.get(TEXTS.get(0));
+    for (String text : TEXTS) {
+      assertEquals(1, knowledgeAnswers.getAnswersCount());
+      Answer answer = knowledgeAnswers.getAnswers(0);
+      assertEquals(text, answer.getFaqQuestion());
+      assertEquals(document.getName(), answer.getSource());
+      assertThat(answer.getAnswer()).contains("Cloud Storage");
+    }
+  }
```

Comment:
Please cleanup the resources (remove the knowledge base) after the test.

Reply:
This is done in the `@After tearDown()` method above, line 77.